mp/src/game/shared/neo/neo_gamerules.cpp

```diff
@@ -1196,11 +1212,196 @@ void CNEORules::GatherGameTypeVotes()
 	m_nGameTypeSelected = mostPopularGameType;
 }
 
-void CNEORules::StartNextRound()
+bool CNEORules::ReadyUpPlayerIsReady(CNEO_Player *pNeoPlayer) const
+{
+	if (!pNeoPlayer) return false;
+
+	const CSteamID steamID = GetSteamIDForPlayerIndex(pNeoPlayer->entindex());
+	return pNeoPlayer->IsBot() || (steamID.IsValid() && m_readyAccIDs.HasElement(steamID.GetAccountID()));
+}
+
+void CNEORules::CheckChatCommand(CNEO_Player *pNeoCmdPlayer, const char *pSzChat)
+{
+	const bool bHasCmds = neo_sv_readyup_lobby.GetBool();
+	if (!bHasCmds || !pNeoCmdPlayer || !pSzChat || pSzChat[0] != '!') return;
+	++pSzChat;
+
+	if (V_strcmp(pSzChat, "help") == 0)
+	{
+		char szHelpText[512];
+		V_sprintf_safe(szHelpText, "Available commands:\n%s",
+					   neo_sv_readyup_lobby.GetBool() ?
+						   "Ready up commands (only available while waiting for players):\n"
+						   "!ready - Ready up yourself\n"
+						   "!unready - Unready yourself\n"
+						   "!overridelimit - Override players amount restriction\n"
+						   "!playersnotready - List players that are not ready\n"
+						 : "");
+		ClientPrint(pNeoCmdPlayer, HUD_PRINTTALK, szHelpText);
+	}
+
+	if (neo_sv_readyup_lobby.GetBool() && m_nRoundStatus != NeoRoundStatus::Idle)
+	{
+		for (const auto pSzCheck : {"ready", "unready", "overridelimit", "playersnotready"})
+		{
+			if (V_strcmp(pSzChat, pSzCheck) == 0)
+			{
+				ClientPrint(pNeoCmdPlayer, HUD_PRINTTALK, "You cannot use this command during a match.");
+				break;
+			}
+		}
+	}
+	else if (neo_sv_readyup_lobby.GetBool() && m_nRoundStatus == NeoRoundStatus::Idle)
+	{
+		const CSteamID steamID = GetSteamIDForPlayerIndex(pNeoCmdPlayer->entindex());
+		if (steamID.IsValid())
+		{
+			const int iThres = neo_sv_readyup_teamplayersthres.GetInt();
+			if (V_strcmp(pSzChat, "ready") == 0)
+			{
+				m_readyAccIDs.Insert(steamID.GetAccountID());
+				ClientPrint(pNeoCmdPlayer, HUD_PRINTTALK, "You are now marked as ready.");
+				const auto readyPlayers = FetchReadyPlayers();
+				if (readyPlayers.array[TEAM_JINRAI] == iThres && readyPlayers.array[TEAM_NSF] == iThres)
+				{
+					UTIL_ClientPrintAll(HUD_PRINTTALK, "All players are ready! Starting soon...");
+				}
+			}
+			else if (V_strcmp(pSzChat, "unready") == 0)
+			{
+				m_readyAccIDs.Remove(steamID.GetAccountID());
+				ClientPrint(pNeoCmdPlayer, HUD_PRINTTALK, "You are now marked as unready.");
+			}
+			else if (V_strcmp(pSzChat, "overridelimit") == 0)
+			{
+				const auto readyPlayers = FetchReadyPlayers();
+				if (readyPlayers.array[TEAM_JINRAI] > iThres && readyPlayers.array[TEAM_NSF] > iThres)
+				{
+					m_bIgnoreOverThreshold = true;
+					ClientPrint(pNeoCmdPlayer, HUD_PRINTTALK, "Overriding threshold, allowing more players.");
+				}
+				else
+				{
+					ClientPrint(pNeoCmdPlayer, HUD_PRINTTALK, "You must go pass the threshold in order to set override.");
```

Comment:
"go past the"

Maybe print the server command used to lower the players per team threshold/target